sklearn/tests/test_docstring_parameters.py

```diff
@@ -0,0 +1,173 @@
+"""
+Test mismatches between docstring and method/function signature.
+"""
+
+# Authors: Joel Nothman <joel.nothman@gmail.com>
+# License: BSD 3 clause
+
+
+# TODO inspect for Cython (see sagenb.misc.sageinspect)
+from __future__ import print_function
+
+import os
+import inspect
+import imp
+import sys
+import sklearn
+import warnings
+from pkgutil import walk_packages
+from nose import SkipTest
+
+
+scikit_learn_path = os.getcwd().replace("/sklearn/tests", "")
+docscrape_path = 'doc/sphinxext/numpy_ext/docscrape.py'
+docscrape = imp.load_source('docscrape',
+                            os.path.join(scikit_learn_path, docscrape_path))
+
+# To capture warnings related to invalid section names in docs
+warnings.simplefilter('always')
+
+
+def get_name(func):
+    parts = []
+    module = inspect.getmodule(func)
+    if module:
+        parts.append(module.__name__)
+    if hasattr(func, 'im_class'):
+        parts.append(func.im_class.__name__)
+    parts.append(func.__name__)
+    return '.'.join(parts)
+
+
+def check_parameters_match(func, doc=None):
```

Comment:
BTW Where should the test go? and also what should the file be named?

Could I move  this method to `utils/testing.py` and add a test to `utils/tests/test_testing.py` ?
Or should  I simply add a file `test_check_parameters_match.py` to the `sklearn/tests/` ?